### DISCUSSION TITLE
Is it possible to use Uncrustify with a different language (AGS Script)

### DISCUSSION THREAD

**Discussion by ericoporto** [uncrustify/uncrustify#4366] [Q&A]
Hi, I would like to use Uncrustify with an unsupported language, namely AGS Script, the scripting language from Adventure Game Studio. It's somewhat similar to C# and C but much much simpler.

If it is possible, I would like to add it to the AGS Editor for formating code. (See https://github.com/adventuregamestudio/ags/issues/2016)

Thanks

**Comment by guy-maurel** [comment]
yes,
you may need to use -l           : Language override: C, CPP, D, CS, JAVA, PAWN, OC, OC+, VALA
`uncrustify  -c <myconfig> -f <myinput> -l CS`
If this is not what are looking for, you need to create some more symbols within uncrustify,
complete the code for your use, provide some tests files to prove it is working well.

**Reply by ericoporto** [reply]
I see, it looks like the code for each language is distributed across the files of Uncrustify itself. I see there's a LANG_ALL in the keywords stuff but some I can see a keyword (`default`) that we don't support.

Is the way to add a new language to include it across all the different files?
